bridges/CarThrottleBridge.php

```diff
@@ -30,24 +78,100 @@ public function collectData()
-            if ($authorDiv) {
-                $item['author'] = $authorDiv[0]->innertext;
-            }
+            $item['author'] = $this->parseAuthor($articlePage);
+
+            $articleElement = $articlePage->find('article')[0];
 
-            $fullArticle = $articlePage->find('article')[0];
             //remove ads
-            foreach ($fullArticle->find('aside') as $ad) {
+            foreach ($articleElement->find('aside') as $ad) {
                 $ad->outertext = '';
-                $fullArticle->save();
             }
 
-            $summary = $fullArticle->find('div.summary')[0];
-            $articleMain = $fullArticle->find('#lbs-content')[0];
+            $summary = $articleElement->find('div.summary')[0];
 
-            $item['content'] = $summary . $articleMain;
+            //remove header so we are left with the article content
+            foreach ($articleElement->find('header') as $found) {
+                $found->outertext = '';
+            }
+
+            //remove comments (registering on carthrottle.com is impossible so the comment sections are empty anyway)
+            foreach ($articleElement->find('#lbs-comments') as $found) {
+                $found->outertext = '';
+            }
+
+            //these are supposed to be hidden
+            foreach ($articleElement->find('.visually-hidden') as $found) {
+                $found->outertext = '';
+            }
+
+            //todo is this necessary?
+            //replace embedded youtube videos with links
+            // foreach ($articleElement->find('iframe') as $found) {
+            //     $iframeUrl = $found->getAttribute('src');
+
+            //     //simply using the existing url doesn't work because RSS-bridge removes double slashes
+            //     if(preg_match('/youtu\.be\/+(\w+)/', $iframeUrl, $matches))
+            //     {
+            //         $youtubeID = $matches[1];
+            //     }
+            //     else {
+            //         continue;
+            //     }
+
+            //     $thumbnailURL = 'https://i.ytimg.com/vi/' . $youtubeID . '/hqdefault.jpg';
+            //     $videoURL = 'https://youtu/be/' . $youtubeID;
+            //     $found->outertext = '<a href="' . $videoURL . '"><img src="' . $thumbnailURL . '" alt="' . $videoURL . '"></a>';
+            // }
+
+            $item['content'] = $summary . $articleElement;
 
             array_push($this->items, $item);
         }
     }
+
+    private function parseAuthor($articlePage)
+    {
+        $authorDivs = $articlePage->find('div address');
+        if (!$authorDivs) {
+            return '';
+        }
+
+        $a = $authorDivs[0]->find('a');
+        if($a){
+            return $a->innertext;
+        }
+        
+        return $authorDivs[0]->innertext;
+    }
+
+    //convert relative url to absolute
+    //https://stackoverflow.com/a/4444490/4671120
+    private function rel2abs($rel, $base)
```

Comment:
we have helper function for this

Reply:
im not using this function anymore, ill remove it